EIPS/eip-5792.md

```diff
@@ -122,26 +136,41 @@ Returns the status of a call batch that was sent via `wallet_sendCalls`. The ide
 type GetCallsParams = string;
 
 type GetCallsResult = {
-  status: 'PENDING' | 'CONFIRMED';
+  batchId: `0x${string}`;
+  chainId?: `0x${string}`;
+  batchStatus: string; // See "Status Codes"
   receipts?: {
     logs: {
       address: `0x${string}`;
       data: `0x${string}`;
       topics: `0x${string}`[];
     }[];
     status: `0x${string}`; // Hex 1 or 0 for success or failure, respectively
-    chainId: `0x${string}`;
     blockHash: `0x${string}`;
     blockNumber: `0x${string}`;
     gasUsed: `0x${string}`;
     transactionHash: `0x${string}`;
   }[];
+  capabilitiesData?: Record<string, any> | undefined;
 };
 ```
 
+##### Status Codes for `batchStatus` field
+
+The purpose of the `batchStatus` field is to provide a short summary of the current status of the batch.
+It provides some off-chain context to the array of inner transaction `receipts`.
+
+| Name      | Description                                                                                                          |
```

Comment:
I like this idea, thoughts @jxom @lukasrosario ?

Reply:
yeah i think this makes sense if we're talking about single chain. would be tough if multichain.

also not clear to me where `DISCARDED` would come up. @forshtat can you share more on how you were thinking this might come up?

Reply:
Yes, it sounds similar to "FAILURE" but the difference I had in mind was that a "FAILURE" means the action failed due to on-chain reverts, and "DISCARDED" means the wallet gave up on trying to send the batch to the network.
Maybe the names do not make the distinction clear enough, or do you think the distinction is not worth the extra status code?

One example could be a batch that specifies a `paymasterService` capability, and given the wallet need to go through these steps before sending anything on-chain:
<img width="756" alt="Screenshot 2024-08-28 at 18 16 29" src="https://github.com/user-attachments/assets/a0255df8-fa3f-47a8-a08d-0cb768faf217">

The wallet cannot block the `wallet_sendCalls()` for all this time, so returns some `identifier` and proceeds with preparing the UserOperation. 
During the preparation phase the `wallet_getCallsStatus` returns PENDING as nothing really happened.
However, if the "Paymaster Service" fails with an error at some stage, wallet could keep retrying to reach it for some time, and the transaction has not technically "failed", but after 2 minutes of trying to reach the "Paymaster Service" it gives up.


Another example is a UserOperation that has some condition in its validation function. Canonical ERC-4337 limits this use-case a lot, but does not completely prevent it. A wallet may want to accept a batch that is not yet valid, check if it has become valid during the next 2 hours, and mark the entire operation as "DISCARDED" after that.

Does this makes sense?